### ISSUE TITLE
Travis/Docker timing

### ISSUE THREAD

**Issue by fentoad72** [dtcenter/METplus#649]
This is a sub-issue of #*Github #639 *.

## Describe the Sub-Issue ##
Add timing statements throughout the code to get an idea of how long each component and subcomponent takes, and whether things can be optimized such that wall-clock time is reduced (currently ~48 minutes).  Tradeoffs include time to download images (currently 1m 35sec) and data volumes vs run time
### Time Estimate ###
*3 dats
### Assignee ###
- [ ] Select **engineer(s)** or **no engineer** required
- [ ] Select **scientist(s)** or **no scientist** required

### Labels ###
- [ ] Select **component(s)**
- [ ] Select **priority**
- [ ] Select **requestor(s)**

### Projects and Milestone ###
- [ ] Review **projects** and select relevant **Repository** and **Organization** ones or add "alert:NEED PROJECT ASSIGNMENT" label
- [ ] Select **milestone** to next major version milestone or "Future Versions"

**Comment by fentoad72** [comment]
I formatted the timing to mm:ss and also added a job_start timestamp in before_script and a job_end timestamp in after_script in travis.yml.  A couple syntax errors and some consistency (test_use_cases, not test_use_case, to match filenames and be consistent).  Left the sequencing alone for now.

I then copied a branch called feature_649a_travisticktockbuild, where docker_setup is called and docker push is commented out.  docker_setup.sh probably doesn't need to be its own stage if we're building in each test_use_cases, and docker_pull in docker_run_metplus.sh doesn't need to be there, but I left them for now and kept the same sequencing (first 5 jobs serial, the rest parallel) for apples to apples.

I'm going to shift my focus to UFS tasks to get ready for the SR-Wx-App tutorial and submit pushes to github (triggering Travis) semi randomly for the next few days before making any other changes (unless a bug comes up) for the most apples-to-apples comparison.  Next week I'll make changes such as those above, then resequence the order of the stages (equally) to see whether we can save some time.

**Comment by fentoad72** [comment]
This link popped up when I logged on to my Travis repo just now:
[https://blog.travis-ci.com/2020-09-11-arm-on-aws](https://blog.travis-ci.com/2020-09-11-arm-on-aws)

It appears that they're upgrading their processor speed.  This would be a good thing if it knocks our 30 minute jobs down to 20, for example.  However, I'm going to keep things the way they are for now, so we continue to have a consistent build environment for the timing issue.  It's good to now they're upgrading, though.

**Comment by fentoad72** [comment]
Timing tests for 11 test runs -- 5 with building the Docker image once and pulling it for the other jobs, 6 with building a fresh Docker image for each job, showed the always-build cases took around 2 minutes less of wall-clock time than the build-once cases.  The actual increase might be slightly more, as I was including two of the use case tests (pytest and test_met_tool_wrapper in the serial tests where they would usually be part of the pool of parallel tests; also I included a "do nothing" step in the place of the build-once subroutine that took 37 seconds to run.  The code has been updated to use python rather than bash as the job manager so there is no need to merge the feature_649a_code (always build) back into develop.
